template/src/main/java/com/asyncapi/model/$$objectSchema$$.java

```diff
@@ -147,8 +147,14 @@ public class {{allName}} {
     {%- if prop.deprecated() %}@Deprecated{% endif %}
     {%- if prop.minLength() or prop.maxLength() or prop.maxItems() or prop.minItems() %}@Size({% if prop.minLength() or prop.minItems() %}min = {{prop.minLength()}}{{prop.minItems()}}{% endif %}{% if prop.maxLength() or prop.maxItems() %}{% if prop.minLength() or prop.minItems() %},{% endif %}max = {{prop.maxLength()}}{{prop.maxItems()}}{% endif %}){% endif %}    
     {%- if prop.pattern() %}@Pattern(regexp="{{prop.pattern() | addBackSlashToPattern}}"){% endif %}
-    {%- if prop.minimum() %}@Min({{prop.minimum()}}){% endif %}{% if prop.exclusiveMinimum() %}@Min({{prop.exclusiveMinimum() + 1}}){% endif %}
-    {%- if prop.maximum() %}@Max({{prop.maximum()}}){% endif %}{% if prop.exclusiveMaximum() %}@Max({{prop.exclusiveMaximum() + 1}}){% endif %}
+    {%- if prop.type() == 'number' and (prop.format() == 'float' or prop.format() == 'double') %}
+        {%- if prop.minimum() %}@DecimalMin({{prop.minimum()}}, inclusive = true){% endif %}{%- if prop.exclusiveMinimum() %}@DecimalMin({{prop.exclusiveMinimum() + 1}}, inclusive = false){% endif %}
```

Comment:
You don't need to add +1 (also below):
![image](https://github.com/user-attachments/assets/a85ff70e-eeef-4502-acab-43e3a1e2804c)